### PR TITLE
feat: redesign player bar to centered-focus layout

### PR DESCRIPTION
**Pull request by rube-de**
## Summary

- Audio player redesigned to the **"Centered Focus"** layout: full-width seek bar on top; transport cluster centered via CSS grid (`grid-cols-[minmax(0,1fr)_auto_minmax(0,1fr)]`) with chapter-prev / rewind pill / 56px emerald play / forward pill / chapter-next; right-side ancillary controls unchanged. Mobile keeps the 2-row layout with chevron icons matching the new defaults.
- **Asymmetric skip defaults**: back 10s, forward 30s (was symmetric 15s). Media Session `seekbackward`/`seekforward` handlers pick them up automatically for lock-screen / Bluetooth remotes.
- **Skip feedback pill**: 700ms `−10s` / `+30s` toast anchored above the player. Monotonic-nonce keyed so sub-ms rapid presses always re-animate; `aria-hidden` so it's not announced per press; state + pending timer cleared when the player becomes invisible.
- **Chapter navigation**: prev uses iPod-style restart-vs-previous (3s threshold); next falls through to `playNext()` on the last chapter with a non-empty queue. aria-label flips between "Next chapter" and "Next episode" to match the actual action. Prev is hidden entirely for queue-only episodes.

## Implementation notes

- Came from a Claude Design handoff bundle (`UT2FybT7tZ8V4ILf38hr1w`, "Centered Focus" variant). Iterated through Codex review (3 correctness fixes in commit 2) and the full `pr-review-toolkit:review-pr` pass (flash nonce, isVisible cleanup, a11y, polish in commit 3).
- Kept the existing `--primary: 160 70% 29%` token rather than the design's 35% — the project intentionally darkened it for WCAG AA (documented in `globals.css`).
- `useMediaQuery` gating around `QueuePanel` mount is preserved: the reviewer-suggested alternative (unconditional mount in both layout wrappers) would render two Popover portals on desktop when open (one anchored to a `display:none` trigger). The hypothetical hydration state-loss the reviewer flagged isn't reachable because the user can't open the panel before JS hydrates.

## Test plan

- [x] `bun run lint` clean
- [x] `bun run build` succeeds
- [x] `bun run test` — 83 player-related tests pass (15 PlayerBar + 68 AudioPlayerContext)
- [x] Verified end-to-end in a browser via `agent-browser` on `localhost:3000`: play/pause, skip `−10s` / `+30s` with flash toast visible and horizontally centered, chapter prev/next, episode switching, dark mode
- [x] Responsive check at ≤767px (mobile 2-row layout) and at ~1000px (narrow desktop — no overlap between transport cluster and right-side controls)
- [ ] Reviewer: confirm no queue-panel regression on desktop (open queue, reorder, clear)
- [ ] Reviewer: confirm hardware media keys / Bluetooth remotes skip 10s / 30s
- [ ] Reviewer: consider follow-up PR for test coverage gaps flagged by `pr-review-toolkit` (chapter prev/next, iPod threshold, skipFlash lifecycle, Media Session defaults — ~8–10 new test cases with `vi.useFakeTimers()`)

## Summary by CodeRabbit

* **New Features**
  * Redesigned audio player UI: full-width seek bar, centered transport row with fixed play button, revamped prev/next and forward pill, and condensed right-side controls.

* **Improvements**
  * Skip defaults changed to Back 10s / Forward 30s; Media Session aligned.
  * Transient skip overlay with re‑animate/reset behavior, clears when player hidden, reduced screen‑reader noise.
  * Smarter chapter-aware prev/next with 3s restart threshold and queue-aware next; ARIA labels updated; mobile layout preserved.

* **Tests**
  * Expanded tests for chapter navigation, skip flash lifecycle, rapid-next, and enablement states.

### CODE REVIEW

File: src/components/audio-player/player-bar.tsx [hunk 1]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
 import { useCurrentChapter } from "@/hooks/use-current-chapter"
 import { useMediaQuery } from "@/hooks/use-media-query"
 
+const REWIND_SECONDS = 10
+const FORWARD_SECONDS = 30
+const PREV_CHAPTER_RESTART_THRESHOLD_SECONDS = 3
+const SKIP_FLASH_DURATION_MS = 700
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: exported SKIP_BACK_SECONDS and SKIP_FORWARD_SECONDS from audio-player-context and imported them into PlayerBar so the API defaults, UI labels, and Media Session handlers share a single source of truth.

File: src/contexts/audio-player-context.tsx [hunk 2]
```diff
@@ -284,7 +284,8 @@ export const AudioPlayerProgressContext = createContext<AudioPlayerProgress | nu
 // Provider
 // ---------------------------------------------------------------------------
 
-const SKIP_SECONDS = 15
+const SKIP_BACK_SECONDS = 10
+const SKIP_FORWARD_SECONDS = 30
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/contexts/audio-player-context.tsx]: Fixed: exported SKIP_BACK_SECONDS and SKIP_FORWARD_SECONDS from audio-player-context and imported them into PlayerBar so the API defaults, UI labels, and Media Session handlers share a single source of truth.

File: src/components/audio-player/player-bar.tsx [hunk 3]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
 import { useCurrentChapter } from "@/hooks/use-current-chapter"
 import { useMediaQuery } from "@/hooks/use-media-query"
 
+const REWIND_SECONDS = 10
+const FORWARD_SECONDS = 30
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: exported SKIP_BACK_SECONDS and SKIP_FORWARD_SECONDS from audio-player-context and imported them into PlayerBar so the API defaults, UI labels, and Media Session handlers share a single source of truth.

File: src/components/audio-player/player-bar.tsx [hunk 4]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+  const hasChapters = chapters != null && chapters.length > 0
+  const canNavigateQueue = queue.length > 0
+  const showNavButtons = hasChapters || canNavigateQueue || chaptersLoading
+
+  const currentChapterIdx = useMemo(() => {
+    if (!hasChapters) return -1
+    let idx = -1
+    for (let i = 0; i < chapters.length; i++) {
+      if (currentTime >= chapters[i].startTime) idx = i
+    }
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: useCurrentChapter now returns { chapter, index } using a single O(log n) binary search; PlayerBar reuses the index and the chapter from the same call instead of running a separate linear scan on every progress update.

File: src/components/audio-player/player-bar.tsx [hunk 5]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+    return idx
+  }, [chapters, currentTime, hasChapters])
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: useCurrentChapter now returns { chapter, index } using a single O(log n) binary search; PlayerBar reuses the index and the chapter from the same call instead of running a separate linear scan on every progress update.

File: src/components/audio-player/player-bar.tsx [hunk 6]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+    }
+    if (canNavigateQueue) {
+      playNext()
+    }
+  }, [hasChapters, chapters, currentChapterIdx, canNavigateQueue, seek, playNext])
+
+  const willAdvanceChapter =
+    hasChapters && currentChapterIdx + 1 < chapters.length
+  const prevLabel = hasChapters ? "Previous chapter" : "Previous episode"
+  const nextLabel = willAdvanceChapter ? "Next chapter" : "Next episode"
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: nextLabel now derives from both willAdvanceChapter and canNavigateQueue — "Next chapter" when a chapter advance is available, "Next episode" when the queue fallback is available, and a generic "Next" when the button is disabled with no action available.

File: src/components/audio-player/player-bar.tsx [hunk 7]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+      const nextIdx = currentChapterIdx + 1
+      if (nextIdx < chapters.length) {
+        seek(chapters[nextIdx].startTime)
+        return
+      }
+    }
+    if (canNavigateQueue) {
+      playNext()
+    }
+  }, [hasChapters, chapters, currentChapterIdx, canNavigateQueue, seek, playNext])
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: added PlayerBar tests covering chapter-prev <3s vs ≥3s threshold and the first-chapter edge case, next-chapter advance vs queue fallback, disabled "Next" state when no action is available, and skip-flash timer lifecycle (fires → clears after 700ms → re-resets on a subsequent skip → clears when the player becomes invisible).

File: src/components/audio-player/player-bar.tsx [hunk 8]
```diff
@@ -29,17 +32,104 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+  const handleSkipForward = useCallback(() => {
+    skipForward(FORWARD_SECONDS)
+    flashSkip("forward", FORWARD_SECONDS)
+  }, [skipForward, flashSkip])
+
+  const handlePrevNav = useCallback(() => {
+    if (hasChapters && currentChapterIdx >= 0) {
+      const current = chapters[currentChapterIdx]
+      const elapsed = currentTime - current.startTime
+      if (elapsed < PREV_CHAPTER_RESTART_THRESHOLD_SECONDS && currentChapterIdx > 0) {
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: added PlayerBar tests covering chapter-prev <3s vs ≥3s threshold and the first-chapter edge case, next-chapter advance vs queue fallback, disabled "Next" state when no action is available, and skip-flash timer lifecycle (fires → clears after 700ms → re-resets on a subsequent skip → clears when the player becomes invisible).

**Comment by rube-de** [comment]
> **Nitpick comments (1)** – src/components/audio-player/player-bar.tsx:60-67: Optional: short-circuit `currentChapterIdx` scan once past `currentTime`...

Fixed: the linear scan was replaced entirely — `useCurrentChapter` now returns `{ chapter, index }` from a single O(log n) binary search that PlayerBar reuses. The short-circuit optimization is superseded by the refactor.

File: src/components/audio-player/player-bar.tsx [hunk 9]
```diff
@@ -29,17 +34,97 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+    if (hasChapters) {
+      const nextIdx = currentChapterIdx + 1
+      if (nextIdx < chapters.length) {
+        seek(chapters[nextIdx].startTime)
+        return
+      }
+    }
+    if (canNavigateQueue) {
+      playNext()
+    }
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:133]: Fixed: handleNextNav and canGoNext now short-circuit while `chaptersLoading` is true, so pressing Next during the chapter fetch is a no-op (button is disabled in the UI) instead of ejecting the user to the next episode. Added a test covering the gating behavior.

File: src/components/audio-player/player-bar.tsx [hunk 10]
```diff
@@ -83,126 +173,157 @@ export function PlayerBar() {
...
-              className="h-8 w-8"
+              variant="secondary"
+              onClick={handleSkipBack}
+              aria-label={`Skip back ${SKIP_BACK_SECONDS} seconds`}
+              title={`Rewind ${SKIP_BACK_SECONDS}s`}
+              className="h-11 w-12 rounded-full border border-border"
             >
-              <SkipBack className="h-4 w-4" />
+              <ChevronsLeft className="h-[22px] w-[22px]" />
             </Button>
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:267]: Fixed: added `size="icon"` so the pill/play buttons use the no-padding icon variant; explicit `h-11 w-12` / `h-14 w-14` remain authoritative via tailwind-merge, and the clickable box is now deterministic.

File: src/components/audio-player/player-bar.tsx [hunk 11]
```diff
@@ -83,126 +173,157 @@ export function PlayerBar() {
...
             <Button
               variant="default"
-              size="icon"
               onClick={togglePlay}
               aria-label={isPlaying ? "Pause" : "Play"}
-              className="h-9 w-9"
+              className="h-14 w-14 shrink-0 rounded-full shadow-md hover:shadow-lg"
             >
               {isBuffering ? (
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:275]: Fixed: added `size="icon"` so the pill/play buttons use the no-padding icon variant; explicit `h-11 w-12` / `h-14 w-14` remain authoritative via tailwind-merge, and the clickable box is now deterministic.

File: src/components/audio-player/player-bar.tsx [hunk 12]
```diff
@@ -83,126 +173,157 @@ export function PlayerBar() {
...
             </Button>
+            <Button
+              variant="secondary"
+              onClick={handleSkipForward}
+              aria-label={`Skip forward ${SKIP_FORWARD_SECONDS} seconds`}
+              title={`Forward ${SKIP_FORWARD_SECONDS}s`}
+              className="h-11 w-12 rounded-full border border-border"
+            >
+              <ChevronsRight className="h-[22px] w-[22px]" />
+            </Button>
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:292]: Fixed: added `size="icon"` so the pill/play buttons use the no-padding icon variant; explicit `h-11 w-12` / `h-14 w-14` remain authoritative via tailwind-merge, and the clickable box is now deterministic.

**Comment by rube-de** [comment]
> **Nitpick comments (2)** — SKIP_FLASH_DURATION_MS drift risk and duplicate `data-testid="current-chapter-title"` across desktop/mobile layouts.

Fixed both:
- `SKIP_FLASH_DURATION_MS` is now exported from `player-bar.tsx`; the skip-flash lifecycle tests import it and derive the advance amounts from it (e.g. `Math.floor(SKIP_FLASH_DURATION_MS * 0.57)`) so the test math tracks the component constant.
- The duplicated `data-testid` is disambiguated to `current-chapter-title-desktop` and `current-chapter-title-mobile` so future `getByTestId` queries won't throw on multiple matches.

File: src/components/audio-player/player-bar.tsx [hunk 13]
```diff
@@ -29,17 +34,102 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+        seek(current.startTime)
+      }
+    }
+  }, [hasChapters, chapters, currentChapterIdx, currentTime, seek])
+
+  const handleNextNav = useCallback(() => {
+    if (hasChapters) {
+      const nextIdx = currentChapterIdx + 1
+      if (nextIdx < chapters.length) {
+        seek(chapters[nextIdx].startTime)
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: `handleNextNav` now tracks the optimistic target via a ref (`nextTargetRef`) so rapid Next presses use the last-advanced index as the base instead of the stale `currentChapterIdx`. An effect clears the ref once live playback catches up, and also clears it if `currentChapterIdx` decreases (e.g. the user picked an earlier chapter from the panel). Added a test that asserts three back-to-back Next clicks seek to chapters[1], [2], [3] in order without `currentTime` needing to refresh.

File: src/components/audio-player/player-bar.tsx [hunk 14]
```diff
@@ -29,17 +34,131 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+        seek(current.startTime)
+      }
+    }
+  }, [hasChapters, chapters, currentChapterIdx, currentTime, seek])
+
+  const handleNextNav = useCallback(() => {
+    if (hasChapters) {
+      const baseIdx =
+        nextTargetRef.current >= 0 ? nextTargetRef.current : currentChapterIdx
+      const nextIdx = baseIdx + 1
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: `handlePrevNav` now clears `nextTargetRef` after each Prev press, so a Next→Prev→Next sequence doesn't inherit a stale optimistic index from the earlier Next. Added a regression test that asserts the second Next seeks to the live chapter's successor (60s), not the would-be index+1 past the optimistic target (180s).

File: src/components/audio-player/player-bar.tsx [hunk 15]
```diff
@@ -29,17 +34,135 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+    // Caught up to (or passed) the optimistic target → clear.
+    if (currentChapterIdx >= nextTargetRef.current) {
+      nextTargetRef.current = -1
+      return
+    }
+    // Live index moved backward (e.g. user picked a chapter from the panel) →
+    // drop the optimistic target so the next nav action starts from the new
+    // live position.
+    if (prev >= 0 && currentChapterIdx < prev) {
+      nextTargetRef.current = -1
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: the catch-up effect now also clears `nextTargetRef` when the live chapter index moves to any value (forward or backward) that isn't our optimistic target. Rapid Next/Prev clicks batch into a single `timeupdate` (the audio element reports only the last seeked value, not intermediate positions), so any real change in `currentChapterIdx` while it's still under the target means the user manually seeked (e.g. chapter-panel pick to chapter 1 while the target was 2). Added a regression test that simulates a mid-sequence chapter-panel pick and asserts the next click seeks to chapter[2] based on the fresh live index, not past it.

File: src/components/audio-player/player-bar.tsx [hunk 16]
```diff
@@ -29,17 +34,137 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+      return
+    }
+    // Live index changed but is still behind the optimistic target: either the
+    // user manually seeked backward or selected an intermediate chapter (via
+    // the chapter panel). Rapid Next/Prev clicks batch into a single
+    // timeupdate, so any real index change under the target means the target
+    // is stale — drop it.
+    if (prev >= 0 && currentChapterIdx !== prev) {
+      nextTargetRef.current = -1
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: refactored to read live audio time directly via a new `getCurrentTime()` on the audio-player API, and compute the chapter index inline from that in both `handleNextNav` and `handlePrevNav`. The optimistic `nextTargetRef`/`prevLiveIdxRef` bookkeeping and the catch-up effect are gone — each press now sees the true current position, so rapid Next, rapid Prev double-press, and any manual seek (same-index chapter-panel picks included) all behave correctly without needing cache-invalidation heuristics.

File: src/components/audio-player/player-bar.tsx [hunk 17]
```diff
@@ -29,17 +34,137 @@ import { BookmarkButton } from "@/components/audio-player/bookmark-button"
...
+    skipForward(SKIP_FORWARD_SECONDS)
+    flashSkip("forward", SKIP_FORWARD_SECONDS)
+  }, [skipForward, flashSkip])
+
+  const handlePrevNav = useCallback(() => {
+    if (hasChapters && currentChapterIdx >= 0) {
+      const current = chapters[currentChapterIdx]
+      const elapsed = currentTime - current.startTime
+      if (elapsed < PREV_CHAPTER_RESTART_THRESHOLD_SECONDS && currentChapterIdx > 0) {
+        seek(chapters[currentChapterIdx - 1].startTime)
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: refactored to read live audio time directly via a new `getCurrentTime()` on the audio-player API, and compute the chapter index inline from that in both `handleNextNav` and `handlePrevNav`. The optimistic `nextTargetRef`/`prevLiveIdxRef` bookkeeping and the catch-up effect are gone — each press now sees the true current position, so rapid Next, rapid Prev double-press, and any manual seek (same-index chapter-panel picks included) all behave correctly without needing cache-invalidation heuristics.

File: src/components/audio-player/player-bar.tsx [hunk 18]
```diff
@@ -83,126 +208,160 @@ export function PlayerBar() {
...
-            <p
-              className="truncate text-xs text-muted-foreground"
-              title={currentEpisode.podcastTitle}
-            >
-              {currentEpisode.podcastTitle}
-            </p>
-            {currentChapter && (
-              <p className="truncate text-xs text-muted-foreground" data-testid="current-chapter-title">
-                {currentChapter.title}
+          {skipFlash.direction === "back" ? "−" : "+"} {skipFlash.seconds}s
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: removed the literal space in the skip-flash JSX so it now renders `−10s` / `+30s` (matching the PR summary). Updated the corresponding test regexes.

**Comment by rube-de** [comment]
> **Nitpick (Non-blocking):** Consider extracting the optimistic next-chapter targeting (`nextTargetRef` / `prevLiveIdxRef` + supporting effects + `baseIdx` math) into a dedicated `useOptimisticChapterTarget` hook.

Answered: the optimistic-target logic was removed entirely in 252fa8c. `handleNextNav` / `handlePrevNav` now read the live audio position via a new `getCurrentTime()` on the audio-player API and compute the chapter index inline via the shared `findChapterIndexAtTime` helper — no refs, no catch-up effects, no invariants to encapsulate. The refactor accomplishes the same goal (PlayerBar focused on UI) by eliminating the bookkeeping rather than hiding it in a hook.

File: src/components/audio-player/player-bar.tsx [hunk 19]
```diff
@@ -26,20 +30,112 @@ import { QueuePanel } from "@/components/audio-player/queue-panel"
...
+    // waiting to see if chapter navigation becomes available.
+    if (chaptersLoading) return
+    if (canNavigateQueue) {
+      playNext()
+    }
+  }, [hasChapters, chapters, chaptersLoading, canNavigateQueue, getCurrentTime, seek, playNext])
+
+  const willAdvanceChapter =
+    hasChapters && currentChapterIdx + 1 < chapters.length
+  const queueAdvanceReady = !chaptersLoading && canNavigateQueue
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: added a `seeked` event listener in `audio-player-context.tsx` that syncs `progress` state immediately after any seek completes. This closes the ~250ms window where `willAdvanceChapter`/`canGoPrev` (derived from state `currentChapterIdx`) could lag behind `handleNextNav`/`handlePrevNav` (already live-time via `getCurrentTime()`), eliminating the action/label mismatch during rapid presses. Preserves the cycle-14 no-optimistic-state invariant — the listener reads directly from the audio element after seek completion.

File: src/contexts/audio-player-context.tsx [hunk 20]
```diff
@@ -970,7 +972,9 @@ export function AudioPlayerProvider({ children }: { children: ReactNode }) {
         audio.currentTime = Math.max(0, Math.min(time, audio.duration || 0))
       },
 
-      skipForward: (seconds = SKIP_SECONDS) => {
+      getCurrentTime: () => audioRef.current?.currentTime ?? 0,
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/contexts/audio-player-context.tsx:975]: Answered: current callers (`handlePrevNav` and `handleNextNav` in `player-bar.tsx:91–126`) gate every `getCurrentTime()` invocation on `hasChapters` — which in turn requires `isVisible` and a loaded `currentEpisode` (player-bar renders `null` otherwise at line 140). Under those gates the audio element is always mounted, so `audioRef.current` is non-null and the `?? 0` fallback never fires. Widening the surface to callers that might invoke without the visibility gate would be a legitimate follow-up (NaN sentinel or a separate `getAudio()` accessor returning `Audio | null`), but that's broader API surface than this PR scope — happy to open a tracker issue if you think it's worth it.

**Comment by rube-de** [comment]
> **Actionable comments posted: 1**
>
> `398-411`: **Guard `mockImplementation` cleanup against assertion failures.**...

Fixed: wrapped the rapid-Next assertion block in try/finally so `mockAPI.seek.mockReset()` always runs, even if an assertion throws. This prevents the mutating implementation from leaking `mockProgress.currentTime` mutations into subsequent tests.

**Comment by rube-de** [comment]
> **Nitpick comments (1)**: Optional: source `SKIP_BACK_SECONDS` / `SKIP_FORWARD_SECONDS` from the real module to prevent drift...

Fixed: switched the `@/contexts/audio-player-context` mock factory to use `importOriginal` and spread `...actual`, so `SKIP_BACK_SECONDS` / `SKIP_FORWARD_SECONDS` now track the real exports. This mirrors the `useCurrentChapter` mock pattern already used at the same file's line 93.

File: src/components/audio-player/player-bar.tsx (hunk 19 above)
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx]: Fixed: added a guard in `handleNextNav` — when live audio has moved past the last chapter but the rendered `willAdvanceChapter` still shows "Next chapter" (the ~250ms timeupdate lag after a natural chapter crossing), the handler now no-ops instead of falling through to `playNext()`. That prevents the unexpected episode skip you flagged: a user clicking a "Next chapter" label never gets an episode transition. On the next timeupdate the label updates to "Next episode" and subsequent clicks work normally. Kept the affordance state-derived and the action live-derived — the guard reconciles them at the decision point without forcing optimistic state or extra render cadence.

File: src/components/audio-player/player-bar.tsx [hunk 21]
```diff
@@ -26,20 +30,117 @@ import { QueuePanel } from "@/components/audio-player/queue-panel"
...
+
+  const handlePrevNav = useCallback(() => {
+    if (!hasChapters) return
+    // Read live audio time so rapid Prev double-press always sees the
+    // just-seeked position instead of a stale `currentTime` from React state.
+    const liveTime = getCurrentTime()
+    const liveIdx = findChapterIndexAtTime(chapters, liveTime)
+    if (liveIdx < 0) return
+    const current = chapters[liveIdx]
+    const elapsed = liveTime - current.startTime
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:99]: Answered: TypeScript 4.4+ supports aliased condition narrowing — assigning a discriminant predicate to a `const` (here, `const hasChapters = chapters != null && chapters.length > 0`) preserves the narrowing when the alias is checked later, including inside callback closures. So after `if (!hasChapters) return`, `chapters` is narrowed to `Chapter[]` for the remainder of the callback body, and `findChapterIndexAtTime(chapters, ...)` / `chapters[liveIdx]` / `chapters.length` all type-check without issue. Verified locally: `bunx tsc --noEmit --strict` exits clean on this PR's HEAD (and `strict: true` in tsconfig.json implies `strictNullChecks: true`). The `hasChapters` form is also used downstream in render (`canGoPrev = hasChapters && currentChapterIdx >= 0`, `showNavButtons = hasChapters || ...`), so keeping a single computed predicate instead of per-callsite guards avoids drift.

File: src/components/audio-player/player-bar.tsx [hunk 22]
```diff
@@ -26,20 +30,117 @@ import { QueuePanel } from "@/components/audio-player/queue-panel"
...
+  const handleNextNav = useCallback(() => {
+    if (hasChapters) {
+      // Read live audio time so rapid Next presses see the real seeked
+      // position — no need to track optimistic state alongside React.
+      const liveTime = getCurrentTime()
+      const liveIdx = findChapterIndexAtTime(chapters, liveTime)
+      const nextIdx = liveIdx + 1
+      if (nextIdx < chapters.length) {
+        seek(chapters[nextIdx].startTime)
+        return
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/components/audio-player/player-bar.tsx:119]: Answered: TypeScript 4.4+ supports aliased condition narrowing — assigning a discriminant predicate to a `const` (here, `const hasChapters = chapters != null && chapters.length > 0`) preserves the narrowing when the alias is checked later, including inside callback closures. So after `if (!hasChapters) return`, `chapters` is narrowed to `Chapter[]` for the remainder of the callback body, and `findChapterIndexAtTime(chapters, ...)` / `chapters[liveIdx]` / `chapters.length` all type-check without issue. Verified locally: `bunx tsc --noEmit --strict` exits clean on this PR's HEAD (and `strict: true` in tsconfig.json implies `strictNullChecks: true`). The `hasChapters` form is also used downstream in render (`canGoPrev = hasChapters && currentChapterIdx >= 0`, `showNavButtons = hasChapters || ...`), so keeping a single computed predicate instead of per-callsite guards avoids drift.
